python/pyarrow/acero.py

```diff
@@ -375,7 +396,13 @@ def _filter_table(table, expression):
     ])
     result = decl.to_table(use_threads=True)
     if is_batch:
-        result = result.combine_chunks().to_batches()[0]
+        batches = result.combine_chunks().to_batches()
+        result = (
+            batches[0]
+            if batches
+            else _empty_record_batch_from_schema(result.schema)
```

Comment:
Do we really need the auxiliary function? Couldn't we just return result (this is already an empty table)

Reply:
Yes. Makes sense. I changed it like this to keep return type same as input type.

Changed it in latest commit.

Reply:
The logic here might not be wrong (I think you are correct about returning same type as input type). We might just make it a bit simpler. cc @raulcd